draft-hendrickson-privacypass-public-metadata-issuance.md

```diff
@@ -43,10 +46,220 @@ TODO Abstract
+
+- Issuer Request URI: A URI to which token request messages are sent. This can
+  be a URL derived from the "issuer-request-uri" value in the Issuer's
+  directory resource, or it can be another Client-configured URL. The value
+  of this parameter depends on the Client configuration and deployment model.
+  For example, in the 'Split Origin, Attester, Issuer' deployment model, the
+  Issuer Request URI might be correspond to the Client's configured Attester,
+  and the Attester is configured to relay requests to the Issuer.
+- Issuer name: An identifier for the Issuer. This is typically a host name that
+  can be used to construct HTTP requests to the Issuer.
+- Issuer Public Key: `pkI`, with a key identifier `token_key_id` computed as
+  described in {{public-issuer-configuration}}.
+- Challenge value: `challenge`, an opaque byte string. For example, this might
+  be provided by the redemption protocol in [AUTHSCHEME].
+- Metadata value: `metadata`, an opaque byte string.
+
+Given this configuration and these inputs, the two messages exchanged in
+this protocol are described below. The constant `Nk` is defined as token type 0x1234 in
+{{PROTOCOL}}.
+
+## Client-to-Issuer Request {#public-request}
+
+The Client first creates an issuance request message for a random value
+`nonce` using the input challenge and Issuer key identifier as follows:
+
+~~~
+nonce = random(32)
+challenge_digest = SHA256(challenge)
+token_input = concat(0x1234, // Token type field is 2 bytes long
+                     nonce,
+                     challenge_digest,
+                     token_key_id)
+blinded_msg, blind_inv =
+  Blind(pkI, metadata, PrepareIdentity(token_input))
+~~~
+
+The PrepareIdentity and Blind functions are defined in
+{{PBLINDRSA}} and {{PBLINDRSA}}, respectively.
+The Client stores the nonce, challenge_digest, and metadata values locally for use
+when finalizing the issuance protocol to produce a token (as described
+in {{public-finalize}}).
+
+The Client then creates a TokenRequest structured as follows:
+
+~~~
+struct {
+  uint16_t token_type = 0x1234; /* Type Public Metadata Blind RSA (2048-bit) */
+  uint8_t truncated_token_key_id;
+  opaque metadata<1..2^16-1>;
+  uint8_t blinded_msg[Nk];
+} TokenRequest;
+~~~
+
+The structure fields are defined as follows:
+
+- "token_type" is a 2-octet integer, which matches the type in the challenge.
+
+- "truncated_token_key_id" is the least significant byte of the `token_key_id`
+  ({{public-issuer-configuration}}) in network byte order (in other words, the
+  last 8 bits of `token_key_id`).
+
+- "metadata" is the opaque metadata value that all `blinded_msg` values are encoded for.
+
+- "blinded_msg" is the Nk-octet request defined above.
+
+The Client then generates an HTTP POST request to send to the Issuer Request
+URI, with the TokenRequest as the content. The media type for this request
+is "application/private-token-request". An example request is shown below:
```

Comment:
Since this is a new TokenRequest structure, we need a new media type.

Reply:
More of a comment for the parent spec, but it seems a little odd we have a different identifier for the structure both in the media type and inside the TokenRequest (token_type) field? Was there a reason the media type is per token type?